src/osversion/VersionTranslator.cpp

```diff
@@ -7664,6 +7945,348 @@ namespace osversion {
+
+        // Rated Supply/Evaporator Fan Power Per Volume Flow Rate 2023
+        newObject.setDouble(insertionIndex + 1, 934.4);
+
+        m_refactored.push_back(RefactoredObjectData(object, newObject));
+        ss << newObject;
+
+      } else if (iddname == "OS:Coil:Cooling:DX:TwoSpeed") {
+
+        // Fields that have been added from 3.6.1 to 3.7.0:
+        // ------------------------------------------------
+        // * Rated High Speed Evaporator Fan Power Per Volume Flow Rate 2017 * 7
+        // * Rated High Speed Evaporator Fan Power Per Volume Flow Rate 2023 * 8
+        // * Rated Low Speed Evaporator Fan Power Per Volume Flow Rate 2017 * 21
+        // * Rated Low Speed Evaporator Fan Power Per Volume Flow Rate 2023 * 22
+
+        auto iddObject = idd_3_7_0.getObject(iddname);
+        IdfObject newObject(iddObject.get());
+
+        for (size_t i = 0; i < object.numFields(); ++i) {
+          if ((value = object.getString(i))) {
+            if (i < 7) {
+              newObject.setString(i, value.get());
+            } else if (i < 19) {
+              newObject.setString(i + 2, value.get());
+            } else {
+              newObject.setString(i + 4, value.get());
+            }
+          }
+        }
+
+        // Rated High Speed Evaporator Fan Power Per Volume Flow Rate 2017
+        newObject.setDouble(7, 773.3);
+        // Rated High Speed Evaporator Fan Power Per Volume Flow Rate 2023
+        newObject.setDouble(8, 934.4);
+
+        // Rated Low Speed Evaporator Fan Power Per Volume Flow Rate 2017
+        newObject.setDouble(21, 773.3);
+        // Rated Low Speed Evaporator Fan Power Per Volume Flow Rate 2023
+        newObject.setDouble(22, 934.4);
+
+        m_refactored.push_back(RefactoredObjectData(object, newObject));
+        ss << newObject;
+
+      } else if (iddname == "OS:AirLoopHVAC:UnitarySystem") {
+
+        // Removed fields
+        // 38 - Maximum Cycling Rate
+        // 39 - Heat Pump Time Constant
+        // 40 - Fraction of On-Cycle Power Use
+        // 41 - Heat Pump Fan Delay Time
+
+        auto iddObject = idd_3_7_0.getObject(iddname);
+        IdfObject newObject(iddObject.get());
+
+        for (size_t i = 0; i < object.numFields(); ++i) {
+          if ((value = object.getString(i))) {
+            if (i < 38) {
+              newObject.setString(i, value.get());
+            } else if (i > 41) {
+              newObject.setString(i - 4, value.get());
+            }
+          }
+        }
+
+        m_refactored.push_back(RefactoredObjectData(object, newObject));
+        ss << newObject;
+
+        auto it = CoilLatentTransitionInfo::findFromParent(coilTransitionInfos, object);
+        if (it != coilTransitionInfos.end()) {
+          if (it->isCurveCreationNeeded()) {
+            IdfObject plfCurve = it->createCurveLinear(idd_3_7_0);
+            ss << plfCurve;
+            m_new.emplace_back(std::move(plfCurve));
+          }
+        }
```

Comment:
When you get to a Unitary System / WAHP, you remove the fields.

You check if a Curve needs to be created, and only if so, you do it.

This ensures that if you have a Unitary System with a Coil:Heating:WaterToAirHeatPump:EquationFit AND a Coil:Cooling:WaterToAirHeatPump:EquationFit, you do create only one unique curve, for both objects to reference.